libs/ui/src/hooks/use_devices.ts

```diff
@@ -78,7 +80,7 @@ export function useDevices({ hardware, logger }: Props): Devices {
 
   useInterval(
     async () => {
-      setBattery(await hardware.readBatteryStatus());
+      setBattery(await makeCancelable(hardware.readBatteryStatus()));
```

Comment:
I would have thought that `useInterval` would clear the interval on unmount. I don't think `hardware.readBatteryStatus()` would take long enough to start before unmount and finish after, especially in test. So I'm surprised this needs the `makeCancelable`. Do you have any insight into why it was needed?

Reply:
It's true that `useInterval` will not _call_ its callback after unmount, but the call to `hardware.readBatteryStatus` necessarily resolves only after the current tick. Since the component may be unmounted in this tick it is possible, _especially_ in a test, that the call to `setBattery` happens after unmount.

Reply:
Makes sense